writeup/src/body.lhs

```diff
@@ -1763,63 +1778,67 @@ telegram-bot       & 729     & 50  & 5   & 0  \\
+behavior. For example, we may not want to share \emph{all} occurrences
+of a variable within a program, but rather only share occurrences of a variable with the same binder.  That
+is, sharing should respect the scope variables. A similar question arises with
+constants -- should all occurrences of the number |1| be shared?
+
+There are a variety of options to customize the sharing behavior of our algorithm.
+One way to do so would allow the definition of a custom oracle
+that is scope-aware. By hashing both the identifier name and its binder,
+we can ensure that variables are not shared over scope boundaries.
+Another option would be consider abstract syntax trees
+that make the binding structure of variables explicit.
 
 \paragraph{Better Merge Algorithm}
 The merging algorithm presented in \Cref{sec:merging} only handles trivial cases.
-Being able to handle the non trivial cases is the current topic of research
-at the time of writing this paper. We wish to better understand the operation
-of merging patches. It seems to share a number of properties from unification theory,
-residual systems, rewriting systems and we would like to look into this 
-in more detail. This would enable us to better pinpoint the role
-that merging plays within our meta-theory, that is, one would expect that it would
-have some resemblance to a pushout as in \cite{Mimram2013}. 
+Being able to merge patches that are not disjoint is the subject of ongoing research.
+The problem seems related to unification,
+residual systems, and rewriting systems. We hope that relating the merging problem to these
+settings might help nail down the necessary conditions for merging to succeed.
+One would expect that it would
+have some resemblance to a pushout, as in pointed out by Mimram and Di Giusto~\cite{Mimram2013}. 
 
 \paragraph{Automatic Merge Strategies}
-Besides improving on the fully generic algorithm, though, we would like to
-have a language to specify domain specific strategies for conflict resolution.
+We would like to
+develop a language to specify domain specific strategies for conflict resolution.
 For instance, whenever the merging tool finds a conflict in the \texttt{build-depends}
-section of a cabal file, it tries sorting the packages alphabetically and keeping
-the ones with the higher version number. Ideally, these rules should be simple to
-write and would allow a high degree of customization.
+section of a cabal file, it might try sorting the packages alphabetically and keeping
+the conflicting packages with the higher version number. Ideally, these rules should be simple to
+write, yet still allow a high degree of customization.
 
 \paragraph{Formalization and Meta-theory}
-We would be happy to engage in a formal verification of our work. This could
+We would be happy to engage in a formal verification of our work.
+%Wouter what does this even mean? Proving the diff-apply laws?
```

Comment:
Yes. We'd also prove closures, proving that there exists an oracle that always chooses
the best patch acording to our notion of `best`, which must be formally stated still, etc...